src/cockpit/389-console/src/lib/security/certificateManagement.jsx

```diff
@@ -239,24 +362,199 @@ export class CertificateManagement extends React.Component {
+                        "warning",
+                        `Error deleting tmp certificate - ${err}`
                     );
                 });
     }
 
+    getCertFiles () {
+        const cmd = [
+            //'/bin/sh', '-c',
+            'find',
+            this.props.certDir,
+            '-type',
+            'f',
+            '-name',
+            '*.pem',
+            '-o',
+            '-name',
+            '*.crt',
+        ];
+        log_cmd("getCertFiles", "creating tmp cert file for importing", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .done((certs_raw) => {
+                    const certs = certs_raw.split(/\r?\n/);
+                    let certNames = [];
+                    for (let i = 0, count = 0; i < certs.length; i++) {
+                        if (certs[i] !== "") {
+                            certNames[count] = certs[i].replace(this.props.certDir + "/", "");
+                            count++;
+                        }
+                    }
+                    certNames.sort();
+
+                    let selectCertName = "";
+                    if (certNames.length > 0) {
+                        selectCertName = certNames[0];
+                    }
+                    this.setState({
+                        availCertNames: certNames,
+                        selectCertName,
+                    })
+                })
+                .fail(err => {
+                    console.log("Failed to get cert file names: ", err);
+                });
+    }
+
+    addCert (isCACert) {
+        this.setState({
+            modalSpinning: true,
+            loading: true,
+        });
+
+        let certType = "certificate";
+        if (isCACert) {
+            certType = "ca-certificate";
+        }
+
+        if ((this.state.certRadioText && this.state.certText !== "") || (this.state.certRadioUpload && this.state.uploadValue)) {
+            // Certificate was copied and pasted.  Need to create a file for import
+            const certFile = this.props.certDir + "/" + this.state.certName + ".tmp";
+            const certText = this.state.certRadioUpload ? this.state.uploadValue : this.state.certText;
+            const create_cert_cmd = [
+                '/bin/sh', '-c',
+                '/usr/bin/echo -e \'' + certText  + '\' > ' + certFile
+            ];
+
+            log_cmd("addCert", "creating tmp cert file for importing: ", create_cert_cmd);
+            cockpit
+                    .spawn(create_cert_cmd, { superuser: true, err: "message" })
+                    .done(() => {
+                        const cmd = [
+                            "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
+                            "security", certType, "add", "--name=" + this.state.certName, "--file=" + certFile
+                        ];
+                        log_cmd("addCert", "Adding cert (tmp): ", cmd);
```

Comment:
Another thing, it doesn't set appropriate flags when I click `Add CA Certificate`. So it's always added to Server Cert table.
See the video (I've also shown the import select issue I mentioned in another comment):
[Screencast from 02-06-2023 01:31:52 PM.webm](https://user-images.githubusercontent.com/5634892/217093292-19a2da75-b1e1-49d0-8380-02e3e05ef603.webm)

Reply:
Ah I keep introducing regressions.  It opened the server cert modal, that's why it did not have the right flags.

Fixed